docs/src/sequencepaxos/storage.md

```diff
@@ -0,0 +1,133 @@
+            self.acc_round = na;
+        }
+
+        fn get_accepted_round(&self) -> Ballot {
+            self.acc_round
+        }
+
+        fn get_entries(&self, from: u64, to: u64) -> Vec<T> {
+            self.log
+                .get(from as usize..to as usize)
+                .unwrap_or(&[])
+                .to_vec()
+        }
+
+        fn get_log_len(&self) -> u64 {
+            self.log.len() as u64
+        }
+
+        fn get_suffix(&self, from: u64) -> Vec<T> {
+            match self.log.get(from as usize..) {
+                Some(s) => s.to_vec(),
+                None => vec![],
+            }
+        }
+
+        fn get_promise(&self) -> Ballot {
+            self.n_prom
+        }
+        ...
+    }
+```
+
+## PersistentStorage
+`PersistentStorage` is a persistent storage implementation that stores the replicated log and the state of OmniPaxos. The struct uses [Commitlog](https://crates.io/crates/commitlog) to store replicated logs, for storing the OmniPaxos state users can choose between [rocksDB](https://crates.io/crates/rocksdb) or [sled](https://crates.io/crates/sled). Users can configure the path to log entries and OmniPaxos state, and storage-related options through `PersistentStorageConfig`. The configuraiton struct features a `default()` constructor for generating default configuration, and the constructor `with()` that takes path and storage options as arguments. We recommend the user to generate the default configuration and then setting the desired fields.
+```rust,edition2018,no_run,noplaypen
+use omnipaxos_core::{
+    sequence_paxos::{SequencePaxos, SequencePaxosConfig},
+};
+use omnipaxos_storage::{
+    persistent_storage::{PersistentStorage, PersistentStorageConfig},
+};
+use commitlog::LogOptions;
+use sled::{Config};
+
+// user-defined configuration
+let my_path = "my_storage"
+let my_log_opts = LogOptions::new(my_path);
+let mut my_sled_opts = Config::new();
+my_sled_opts.path(self, my_path);
+
+// generate default configuration and set user-defined configuration
+let mut my_config = PersistentStorageConfig::default();
+my_config.set_path(my_path);
+my_config.set_commitlog_options(my_logopts);
+my_config.set_sled_options(my_sled_opts);
+```
+The same configuration can also be done with the constructor that takes arguments:
+```rust,edition2018,no_run,noplaypen
+let my_path = "another_storage"
+let my_logopts = LogOptions::new(my_path);
+let mut my_sled_opts = Config::new();
+my_sled_opts.path(my_path);
+my_sled_opts.new(true);
+
+// create configuration with given arguments
+let my_config = PersistentStorageConfig::with(my_path, my_logopts, my_sled_opts);
+```
+## Selecting storage for OmniPaxos state
+`PersistentStorage` is configured to use `sled` by default, users can enable `rocksDB` as storage by adding `--features rocksdb` to their cargo command. The example below runs tests with rocksDB enabled:
+
+```no_run,noplaypen
+cargo test --features rocksdb
+```
```

Comment:
Show an example of how to import omnipaxos_storage with the rocksdb feature in `Cargo.toml` instead.